client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.client.hotrod.event;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.test.RemoteCacheManagerCallable;
+import org.infinispan.client.hotrod.test.SingleHotRodServerTest;
+import org.testng.annotations.Test;
+
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.withRemoteCacheManager;
+import static org.infinispan.test.TestingUtil.detectThreadLeaks;
+
+@Test(groups = "functional", testName = "client.hotrod.event.ClientListenerLifecycleTest")
+public class ClientListenerLeakTest extends SingleHotRodServerTest {
+
+   public void testNoLeaksAfterShutdown() {
+      withRemoteCacheManager(new RemoteCacheManagerCallable(getRemoteCacheManager()) {
+         @Override
+         public void call() {
+            RemoteCache<Integer, String> remote = rcm.getCache();
+            EventLogListener<Integer> eventListener = new EventLogListener<>();
+            remote.addClientListener(eventListener);
+            eventListener.expectNoEvents();
+            remote.put(1, "one");
+            eventListener.expectOnlyCreatedEvent(1, cache());
+         }
+      });
+      detectThreadLeaks(".*Client-Listener.*");
```

Comment:
Won't this fail if there's another test running in parallel and using client listeners?